exasol/ds/sandbox/lib/run_create_vm.py

```diff
@@ -50,18 +50,20 @@ def run_create_vm(
     If anything goes wrong the cloudformation stack of the EC-2 instance will be removed.
     For debuging you can use the available debug commands.
     """
-    source_ami_id = ec2_source_ami or source_ami_id_with_logging(
-        aws_access,
-        configuration.source_ami_filters,
-        LOG,
+    finder = AmiFinder(aws_access, LOG)
+    source_ami = (
+        finder.unique({"image-id": ec2_source_ami}) if ec2_source_ami
+        else finder.latest(configuration.source_ami_filters)
     )
+    if not source_ami:
+        return
```

Comment:
If I understand this correctly, we basically log here in finder only and then rerun with NOn, however the function we are in, should create a ec2 instance, if this is not possible because we don't find an ami or can't find a unique ami, we should raise an exception. So, the whole logic here is questionable.